syntax = "proto3";
package user;
service UserService {
  rpc Register(User) returns (EmptyMessage);
  rpc GetById(UserId) returns (UserResponse);
  rpc Update(User) returns (EmptyMessage);
  rpc Delete(UserId) returns (EmptyMessage);
}

message EmptyMessage {
  optional string error_message = 1;
  optional int32 error_code = 2;
}

message User {
  string id = 1;
  string first_name = 2;
  string last_name = 3;
  string home_address = 4;
  string gender = 5;
  optional bool is_featured = 6;
}

message UserResponse {
  string id = 1;
  string first_name = 2;
  string last_name = 3;
  string home_address = 4;
  string gender = 5;
  optional bool is_featured = 6;
  optional string error_message = 7;
  optional int32 error_code = 8;
}

message UserId {
  string id = 1;
}